spec/models/affiliation_identifier_spec.rb

```diff
@@ -83,138 +69,18 @@
-                "subj" => {},
-                "obj" => {},
-              },
-            },
-          }
-
-          expect(Rails.logger).to receive(:info).with("[Event Data] https://doi.org/10.1234/example-doi is_authored_at https://ror.org/02catss52 pushed to Event Data service.")
-
-          stub_request(:post, push_url).
-            with(
-              body: expected_data.to_json,
-              headers: {
-                "Authorization" => "Bearer example_admin_token",
-                "Content-Type" => "application/vnd.api+json",
-                "Accept" => "application/vnd.api+json; version=2",
-              },
-            ).
-            to_return(status: 200, body: { "data" => { "id" => "example_id" } }.to_json, headers: {})
-
-          AffiliationIdentifier.push_item(item)
-        end
-
-        it "skips pushing if STAFF_ADMIN_TOKEN is not present" do
-          allow(ENV).to receive(:[]).with("STAFF_ADMIN_TOKEN").and_return(nil)
-          expect(Maremma).not_to receive(:post)
-
-          AffiliationIdentifier.push_item(item)
-        end
-
-        it "returns 409 for already pushed events" do
-          allow(Maremma).to receive(:post).and_return(OpenStruct.new(status: 409,
-                                                                     body: { "data" => { "id" => "example_id" } }))
-          push_url = "https://fake.lagattino.com/events"
-          expected_data = {
-            "data" => {
-              "type" => "events",
-              "attributes" => {
-                "messageAction" => "create",
-                "subjId" => "https://doi.org/10.1234/example-doi",
-                "objId" => "https://ror.org/02catss52",
-                "relationTypeId" => "is_authored_at",
-                "sourceId" => "datacite_affiliation",
-                "sourceToken" => "DATACITE_AFFILIATION_SOURCE_TOKEN",
-                "occurredAt" => "2023-01-05T12:00:00Z",
-                "timestamp" => Time.zone.now.iso8601,
-                "license" => "https://creativecommons.org/publicdomain/zero/1.0/",
-                "subj" => {},
-                "obj" => {},
-              },
-            },
-          }
-
-          expect(Rails.logger).to receive(:info).with("[Event Data] https://doi.org/10.1234/example-doi is_authored_at https://ror.org/02catss52 already pushed to Event Data service.")
-
-          stub_request(:post, push_url).
-            with(
-              body: expected_data.to_json,
-              headers: {
-                "Authorization" => "Bearer example_admin_token",
-                "Content-Type" => "application/vnd.api+json",
-                "Accept" => "application/vnd.api+json; version=2",
-              },
-            ).
-            to_return(status: 200, body: { "data" => { "id" => "example_id" } }.to_json, headers: {})
-
-          AffiliationIdentifier.push_item(item)
-        end
-
-        it "returns 500 when there is error while pushing an event" do
```

Comment:
I know you are still working on this, but can we keep these tests which are testing the different response code for different type of requests?

Reply:
@ash

> I know you are still working on this, but can we keep these tests which are testing the different response code for different type of requests?

Thank you for the feedback. Checking response statuses will now be the responsibility of the worker and not the model. As such, these specs will be rewritten when I write the specs for the worker.